tests/test_jwt_utils.py

```diff
@@ -37,12 +37,11 @@ def test_did_not_find_matching_jwk(
         with pytest.raises(Unauthorized):
             decode_jwt("x")
         get_matching_jwk_mock.assert_called_once_with("x")
-        assert "Failed decoding JWT with following details" in caplog.text
+        assert "Failed to find matching jwk" in caplog.text
 
     @patch("lbz.jwt_utils.get_matching_jwk", return_value={})
```

Comment:
You should fix the incorrect assumption here instead of trying to improve code 😉 

_And the same in the above test._